spring-integration-file/src/main/java/org/springframework/integration/file/FileWritingMessageHandler.java

```diff
@@ -101,6 +105,9 @@
 public class FileWritingMessageHandler extends AbstractReplyProducingMessageHandler
 		implements Lifecycle, MessageTriggerAction {
 
+	private static final boolean nioFilesPresent = ClassUtils.isPresent("java.nio.file.Files",
+			FileWritingMessageHandler.class.getClassLoader());
```

Comment:
I think after rebase to current master this change to the `FileWritingMessageHandler` should go from your PR.

From other side looks like I can merge it already!